### PR TITLE
Implement a checkstyle github action

### PR DESCRIPTION
**Pull request by KenwoodFox**
Ideally, this should check if the style guidelines in our code are being met!

### CODE REVIEW

**Comment by KenwoodFox** [comment]
@rmessinger are you good with github actions or CI in java? Id like to have a style checker but I'm not sure how :c

File: .github/workflows/frcbuild.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: FRC Build & Test
-on: [push, pull_request]
```
Comment (rmessinger) [inline .github/workflows/frcbuild.yml:2]: I find running the build step on pushes to be pretty useful

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 2]
```diff
@@ -13,6 +13,9 @@
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.robot.Constants;
 
+
```
Comment (rmessinger) [inline src/main/java/frc/robot/subsystems/Intake.java:16]: whoa now

Comment (KenwoodFox) [reply src/main/java/frc/robot/subsystems/Intake.java:16]: I was trying to get checkstyle to call me out on it :c IDK how to make it work~

Comment (rmessinger) [reply src/main/java/frc/robot/subsystems/Intake.java]: ahhh I see

Comment (KenwoodFox) [reply src/main/java/frc/robot/subsystems/Intake.java]: I think it works properly now :3 it tags and comments every style error just the way id expect it to.

**Comment by KenwoodFox** [comment]
This needs more work but ill work on it later.

**Pull request merged**